modules/electromagnetics/src/bcs/VectorEMRobinBC.C

```diff
@@ -0,0 +1,193 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "VectorEMRobinBC.h"
+#include "ElectromagneticEnums.h"
+#include "Function.h"
+#include <complex>
+
+registerMooseObject("ElectromagneticsApp", VectorEMRobinBC);
+
+InputParameters
+VectorEMRobinBC::validParams()
+{
+  InputParameters params = VectorIntegratedBC::validParams();
+  params.addClassDescription("First order Robin-style Absorbing/Port BC for vector variables.");
+  params.addParam<FunctionName>(
+      "beta", 1.0, "Scalar waveguide propagation constant (usually some k, k0).");
+  MooseEnum component("real imaginary");
+  params.addParam<MooseEnum>(
+      "component", component, "Variable field component (real or imaginary).");
+  params.addRequiredCoupledVar("coupled_field", "Coupled field variable.");
+  params.addParam<FunctionName>("real_incoming", 0.0, "Real incoming field vector.");
+  params.addParam<FunctionName>("imag_incoming", 0.0, "Imaginary incoming field vector.");
+  MooseEnum mode("absorbing port", "port");
+  params.addParam<MooseEnum>(
+      "mode",
+      mode,
+      "Mode of operation for VectorEMRobinBC. Can be set to 'absorbing' or 'port' "
+      "(default: 'port').");
+  return params;
+}
+
+VectorEMRobinBC::VectorEMRobinBC(const InputParameters & parameters)
+  : VectorIntegratedBC(parameters),
+
+    _beta(getFunction("beta")),
+
+    _component(getParam<MooseEnum>("component")),
+
+    _coupled_val(coupledVectorValue("coupled_field")),
+    _coupled_var_num(coupled("coupled_field")),
+
+    _inc_real(getFunction("real_incoming")),
+    _inc_imag(getFunction("imag_incoming")),
+
+    // Value of complex j (can't use _j or _i due to MOOSE basis fxn conventions)
+    _jay(0, 1),
```

Comment:
I wonder if you should just have a namespaced global static const like `EM::jay` or something

Reply:
Or if you did that you could actually call it `EM::j`

Reply:
or `EM::i`

Reply:
Added `utils/ElectromagneticConstants.h` in a forthcoming commit so that there's a central place for mu_0, eps_0, c and others, which should simplify code a bit more as well. Thanks for this nudge!